packages/uicore/src/components/Select/BiLevelSelect.tsx

```diff
@@ -0,0 +1,322 @@
+    | 'itemRenderer'
+    | 'onItemSelect'
+    | 'query'
+    | 'selectedItem'
+    | 'items'
+    | 'activeItem'
+    | 'onActiveItemChange'
+  > {
+  inputValueRender?: BiLevelProps['inputValueRenderer']
+  itemRenderer?: BiLevelProps['itemRenderer']
+  onChange?: BiLevelProps['onItemSelect']
+  value?: BiLevelProps['selectedItem']
+  size?: SelectSize
+  items: BiLevelProps['items'] | (() => Promise<BiLevelProps['items']>)
+  allowCreatingNewItems?: boolean
+  name?: string
+  whenPopoverClosed?: (node: HTMLElement) => void
+  addClearBtn?: boolean
+  usePortal?: boolean
+  popoverClassName?: string
+  onQueryChange?: BiLevelProps['onQueryChange']
+  addTooltip?: boolean
+  borderless?: boolean
+  loadingItems?: boolean
+}
+
+function getIconSizeFromSelect(size: SelectSize = SelectSize.Medium) {
+  let iconSize = 16
+  if (size === SelectSize.Small) {
+    iconSize = 12
+  } else if (size === SelectSize.Large) {
+    iconSize = 18
+  }
+  return iconSize
+}
+
+export function defaultItemRenderer(
+  item: SelectWithBiLevelOption,
+  props: IItemRendererProps,
+  size: SelectSize = SelectSize.Medium
+): JSX.Element | null {
+  if (!props.modifiers.matchesPredicate) {
+    return null
+  }
+
+  if (item.value === Loading) {
+    return (
+      <li key="loading" className={cx(css.menuItem, css.loading)}>
+        Loading results...
+      </li>
+    )
+  }
+
+  return (
+    <li
+      key={item.value.toString()}
+      className={cx(
+        css.menuItem,
+        {
+          [css.active]: props.modifiers.active,
+          [css.disabled]: props.modifiers.disabled
+        },
+        { [css.menuItemSizeSmall]: size === SelectSize.Small },
+        { [css.menuItemSizeLarge]: size === SelectSize.Large }
+      )}
+      onClick={props.handleClick}>
+      {item.icon ? <Icon size={getIconSizeFromSelect(size)} {...item.icon} /> : null}
+      <Text className={css.menuItemLabel} lineClamp={1}>
+        {item.label}
+      </Text>
+    </li>
+  )
+}
+
+export function createNewItemFromQuery(query: string): SelectWithBiLevelOption {
+  return { label: query, value: query }
+}
+
+export function BiLevelSelect(props: BiLevelSelectProps): React.ReactElement {
+  const [query, setQuery] = React.useState('')
+  const [loading, setLoading] = React.useState(false)
+  const [items, setItems] = React.useState(Array.isArray(props.items) ? props.items : [])
+  const {
+    onChange,
+    value,
+    size,
+    itemRenderer,
+    addClearBtn,
+    whenPopoverClosed,
+    popoverClassName = '',
+    resetOnSelect = true,
+    resetOnClose = true,
+    addTooltip = false,
+    borderless = false,
+    loadingItems,
+    ...rest
+  } = props
+  const [item, setItem] = React.useState<SelectWithBiLevelOption | undefined | null>(undefined)
+
+  const showClearBtn =
+    !!addClearBtn &&
+    value !== null &&
+    value !== undefined &&
+    (value as any) !== '' &&
+    value?.value !== undefined &&
+    value?.value !== null &&
+    value?.value !== ''
```

Comment:
can we update this to `!!(addClearBtn && value?.value)`

Reply:
addressed